File: driver/src/main/java/com/glimmer/carrybport/ui/activity/SplashActivity.java
package com.glimmer.carrybport.ui.activity;

import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.view.WindowManager;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.widget.ImageView;

import com.glimmer.carrybport.BuildConfig;
import com.glimmer.carrybport.MyApplication;
import com.glimmer.carrybport.R;
import com.sky.Common;
import com.sky.utils.JumpAct;
import com.sky.utils.SPUtils;

/**
 * Created by ASUS on 2017/6/7.
 */
public class SplashActivity extends AppCompatActivity {
    public Boolean flag;//是否首次运行，true代表首次
    private int TIME = 1000;//handler延迟时间

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        //requestWindowFeature(Window.FEATURE_NO_TITLE);//去掉标题栏,在v7中AppCompatActivity下无用
        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);//去掉信息栏
        super.onCreate(savedInstanceState);
        ImageView view = new ImageView(this);
        view.setBackgroundResource(R.mipmap.bg_splash);
        setContentView(view);
        if (BuildConfig.DEBUG) MyApplication.getInstance().showToast("测试环境");
        flag = (Boolean) SPUtils.getInstance().get(Common.ISFIRST, true);
        //加载动画
        AlphaAnimation animation = new AlphaAnimation(0.5f, 1f);
        animation.setDuration(TIME);
        view.startAnimation(animation);
        animation.setAnimationListener(new Animation.AnimationListener() {

            @Override
            public void onAnimationStart(Animation animation) {
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                new Handler().postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        // 判断程序与第几次运行，如果是第一次运行引导页面
                        JumpAct();
                    }
                }, TIME);
            }
        });
    }

    private void JumpAct() {
        if (MyApplication.getInstance().getObject(Common.ISFIRST, true)) {
            JumpAct.jumpActivity(this, GuideActivity.class);
//        } else if (MyApplication.getInstance().getUsertOnline()){
//            JumpAct.jumpActivity(this, MainActivity.class);
        }else{
            JumpAct.jumpActivity(this, LoginActivity.class);
        }
        finish();
    }
}

File: app/src/main/java/com/sky/oa/activity/SplashActivity.kt
package com.sky.oa.activity

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.WindowManager
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.sky.oa.AppConstants
import com.sky.oa.R
import com.sky.oa.adapter.GuideContoler
import com.sky.sdk.utils.SPUtils

/**
 * Created by SKY on 2017/3/6.
 * 引导页面
 */
class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)//去掉信息栏
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        initData()
    }

    /**
     * 使用写好的库初始化引导页面
     */
    private fun initData() {
        val contoler = GuideContoler(this)
        val imgIds = intArrayOf(R.mipmap.ic_landing, R.mipmap.ic_bg_01, R.mipmap.ic_bg_02)
        val view = LayoutInflater.from(this).inflate(R.layout.pager_end, null)
        contoler.init(imgIds, view)
        view.findViewById<Button>(R.id.btBegin).setOnClickListener {
            SPUtils.getInstance(AppConstants.ISFIRST).put(AppConstants.ISFIRST, false)
            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
//            overridePendingTransition(R.anim.in_from_right, R.anim.out_to_left);
            finish()
        }
    }
}
